pkg/fileutils/fileutils.go

```diff
@@ -294,14 +295,18 @@ func EnsureParentDirectoryExist(fileName string) error {
 	return EnsureDirectoryExists(destinationDir)
 }
 
-// EnsureDirectoryExists check if the passed directory exists or not, and if
-// it doesn't exist, create it using 0700 as permissions bits
+// EnsureDirectoryExists checks whether the directory exists and has the
+// expected 0700 permissions.
+// It will create the directory or update the permissions as needed
 func EnsureDirectoryExists(destinationDir string) error {
-	if _, err := os.Stat(destinationDir); os.IsNotExist(err) {
-		err = os.MkdirAll(destinationDir, 0o700)
-		if err != nil {
-			return err
-		}
+	fileInfo, err := os.Stat(destinationDir)
+	if os.IsNotExist(err) {
+		return os.MkdirAll(destinationDir, 0o700)
+	} else if err != nil {
+		return err
+	}
+	if fileInfo.Mode().Perm() != fs.FileMode(0o700) {
+		return os.Chmod(destinationDir, 0o700) // nolint:gosec
```

Comment:
Please don't do that. It could have unintended consequences, and it goes beyond ensuring that the directory exists.

Reply:
yep, found out the tests were failing. Rolled back.